### ISSUE TITLE
Add support for OpenSSL3x. OpenSSL 1.1.1 went end of life was 11th September 2023.

### ISSUE THREAD

**Issue by sysfu** [EpicCash/epic#134]
**Describe the bug**
OpenSSL 1.1.1 [went EoL](https://www.openssl.org/blog/blog/2023/06/15/1.1.1-EOL-Reminder/) on the 11th of September 2023. The epic server binary will not run on gentoo linux with current versions of OpenSSL libraries installed.

**To Reproduce**
Download and extract latest epic server binary, then run it. Epic binary complains:  `epic: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory
`
**Expected behavior**
epic server binary supports current version(s) of SSL.

**Required data**
 - Node/Wallet/Miner log as text or screenshot
 - Config toml file or command line 
 - OS: [e.g. Windows]
 - Rust Version
 - Cargo Version

**Additional context**
Current version openssl libraries on gentoo is `dev-libs/openssl-3.0.13-r2-17:0/3`

```
ldconfig -p | grep libssl
        libssl.so.3 (libc6,x86-64) => /usr/lib64/libssl.so.3
        libssl.so.3 (libc6) => /usr/lib/libssl.so.3
        libssl.so (libc6,x86-64) => /usr/lib64/libssl.so
        libssl.so (libc6) => /usr/lib/libssl.so
```

**Comment by who-biz** [comment]
Agreed that this should be changed.  However, the most common distributions run by users of Linux are Ubuntu 20.04 and 22.04.  These use different versions of `openssl`. Until Ubuntu 20.04 migrates to 3.0, or hits EOL, we are unfortunately stuck with this situation.

Not Ubuntu's first time creating a situation like this. `glibc` also had an update that broke backwards compat for Ubuntu 16.04 & Debian 9, some years ago.  We are lucky that this is only a library, and does not require separate compiled binaries.
